controllers/grid/settings/user/UserGridHandler.php

```diff
@@ -203,7 +214,17 @@ protected function loadData($request, $filter)
         $collector->limit($rangeInfo->getCount());
         $collector->offset($rangeInfo->getOffset() + max(0, $rangeInfo->getPage() - 1) * $rangeInfo->getCount());
         $iterator = $collector->getMany();
-        return new VirtualArrayIterator(iterator_to_array($iterator, true), $totalCount, $rangeInfo->getPage(), $rangeInfo->getCount());
+        $users = iterator_to_array($iterator, true);
+        foreach ($users as $user) {
+            $roles = array();
+            $userGroupsIterator = Repo::userGroup()->userUserGroups($user->getId(), $context->getId());
+            $userGroups = iterator_to_array($userGroupsIterator);
+            foreach ($userGroups as $userGroup) {
+                $roles[] = $userGroup->getLocalizedName();
+            }
```

Comment:
Not sure if you like, but you might use a functional approach (either way are ok for us):

```php
$userGroupsIterator->map(fn (UserGroup $userGroup) => $userGroup->getLocalizedName())->join(', ');
```